### ISSUE TITLE
BeautifulSoup4 updated crashes ICDrama

### ISSUE THREAD

**Issue by CLFNole** [aznhusband/aznhusband.github.io#58]
Just noticed that the Kodi 18 (not 19) update to BeautifulSoup4 causes ICDrama to stop working. Just wanted to let you know.

**Comment by boniface88** [comment]
I have a similar problem with Kodi 18.8. Some updates by Kodi library crashes Icdrama (not sure which one). Now icdrama doesn’t work for me.

**Comment by jcheong-1** [comment]
Same for me on Kodi 18, I wonder if someone on Kodi 19 can chime in to see if there's any issues for them.

**Comment by jcheong-1** [comment]
Further to my last comment, I took the plunge to upgrade to Kodi 19, icdrama addon is not yet compatible with Kodi 19 at this point, along with most of my other addons.

**Comment by boleanly** [comment]
It seems like the new version of BeautifulSoup 4 In the Kodi 18 repo isn't compatible with the version of Soupsieve in the same repo. BS4 throws an error saying Soupsieve isn't installed even though a version of it is. Kodi 19's BS4 and Soupsieve aren't affected though.

You can use Soraxis's branch for a Kodi 19 compatible version here at #55

**Comment by nahcyrag** [comment]
i dont know what got updated and causing ICdrama not to work. I deleted everything including Kodi or you can do clear data and reinstall ICdrama disable auto-update on this addon and everything work again. I also disabled auto-update on Beautifulsoup and Soupsieve.

**Comment by boniface88** [comment]
Yes, I deleted Beautifulsoup and Soupsieve, along with icdrama, then reinstalled icdrama.  Now icdrama is working again.  Yes, make sure you turn off the auto-update on those dependencies.  Thank you for your solution!

**Comment by cdkao** [comment]
I hope Asianhusband can do the soup update soon because my wife is not happy that it's been down for this long.  No hanky panky for me.

**Comment by nahcyrag** [comment]
> I hope Asianhusband can do the soup update soon because my wife is not happy that it's been down for this long. No hanky panky for me.
just uninstall 19 and sideload 18 again. follow then follow my instruction above and you will be back on icdrama at no time. It has been working for my wife.

**Comment by cdkao** [comment]
> > I hope Asianhusband can do the soup update soon because my wife is not happy that it's been down for this long. No hanky panky for me.
> > just uninstall 19 and sideload 18 again. follow then follow my instruction above and you will be back on icdrama at no time. It has been working for my wife.

**Comment by cdkao** [comment]
> > just uninstall 19 and sideload 18 again. follow then follow my instruction above and you will be back on icdrama at no time. It has been working for my wife.

Thanks!  I hope this addon didn't die as we don't hear back from this dev anymore.

**Comment by xalagaco** [comment]
Icdrama stopped working even on 18.9 for a few days now. It's broken everywhere, I'm starting to lose hope.

**Comment by cdkao** [comment]
> Icdrama stopped working even on 18.9 for a few days now. It's broken everywhere, I'm starting to lose hope.

Read above for the temporary fix. Goodluck!
